### ISSUE TITLE
List of improvements

### ISSUE THREAD

**Issue by hhsnopek** [NYTimes/pourover#54]
Any thoughts on these improvements?
- [x] Use node's `package.json` to handle deps and devDeps
- [x] Use [mocha](//mochajs.org/) for testing
- [x] Add [travis](//travis-ci.org) support for automated testing
- [x] Use npm scripts for building/testing (`npm build` & `npm test`)
- [ ] Update inline documentation to [jsdoc](//github.com/jsdoc3/jsdoc) (still can use docco for generation, I believe)

These should be looked at prior to merging #47 as the PR contains changes that are unnecessary if any of these improvements are agreed upon.

I'll PR these based on the current master for your review.

**Comment by TomNeyland** [comment]
Thats quite a laundry list and rather than hold #47 up waiting on all of that, I would suggest that those changes be made incrementally. Each of those are good changes but they can easily stand on their own as individual pull requests.

Some of the changes in #47 would definitely be unnecessary if all of what you are proposing was already said and done, but given that #47 is ready to go and represents a step in the right direction, I don't necessarily see a convincing reason to wait an undetermined amount of time on the above. And again I think some of those items would be better suited as individual PRs.

**Comment by hhsnopek** [comment]
I agree, but if these changes were to be merged in they would undo the work that's currently done. I'd rather not create the git history, only to undo it in a week or so.

I see 2 PR's for these changes. The first will complete points one through four. The reasoning is that as I'm migrating the tests over to mocha, `pourover.js` itself isn't ready for node usage out of the box. That being said I'll be updating `pourover.js` to be out of the box ready for bower, component, node, and the web. Once the tests and `pourover.js` are passing we can cute a new release, considering it will do exactly what our original task was of adding bower, component and node support. The second PR I will improve documentation, point five.

All these PRs won't break anything currently using pourover, it will just be a major version bump to `1.1.0`  from the first PR and then a minor version bump to `1.1.1`. **Note:** The versioning is based off of [semver](//semver.org/).

**Comment by hhsnopek** [comment]
@TomNeyland I'm going to rebase this onto your fork/branch and send you the PR

**Comment by hhsnopek** [comment]
@TomNeyland I have completed points 1, 2 & 4 and have sent you a PR: https://github.com/TomNeyland/pourover/pull/4. Unless we'd like these changes to not be apart of your PR.

## Notes:
- All tests are passing, I've changed the release number to `1.1.0` as we added "functionality in a backwards-compatible manner".
- Publish to Travis will need to be taken care of by the maintainers of NYTimes.
- Publish to NPM will also need to be taken care of by the maintainers of NYTimes.
- Point 5 will be done in a separate PR and along with this I will add the Travis Badges.

**Comment by tiffehr** [comment]
Am I right in that the recommendations listed here are present in some form in @hhsnopek's version?  If so, I'll close the Issue with a pointer over to that repo for people who need these additions.

**Comment by hhsnopek** [comment]
@tiffehr Yeah, be sure to checkout my community version of pourover which close many of the open issues in this repo. Also I've opened a gitter channel and hold bower/npm ownership atm. It would be awesome if a lot of the upstream changes from my fork could get merged in. Lemme know what you think!

**Comment by hhsnopek** [comment]
Considering that the community edition has these changes, and I'll be mirroring the changes from the repo with other changes. You could close #47

**Comment by tiffehr** [comment]
Do you want to preserve this history/checklist for your own repo in some fashion?

**Comment by hhsnopek** [comment]
I have issues open currently for them, this list was a reference to what was occurring in my fork :)
